tests/transforms/test_clifford_t/test_cliffordt_transform.py

```diff
@@ -0,0 +1,303 @@
+        """Test Clifford checker operation for gate"""
+        assert check_clifford_t(op) == res
+
+    @pytest.mark.parametrize(
+        ("circuit, max_depth"),
+        [(circuit_1, 1), (circuit_2, 0), (circuit_3, 0), (circuit_4, 1), (circuit_5, 0)],
+    )
+    def test_decomposition(self, circuit, max_depth):
+        """Test decomposition for the Clifford transform."""
+
+        with qml.tape.QuantumTape() as old_tape:
+            circuit()
+        print(circuit)
+        [new_tape], tape_fn = clifford_t_decomposition(
+            old_tape, max_depth=max_depth, approximate_set=_APPROX_SET
+        )
+
+        assert all(
+            any(
+                (
+                    isinstance(op, gate) or isinstance(getattr(op, "base", None), gate)
+                    for gate in _CLIFFORD_PHASE_GATES
+                )
+            )
+            for op in new_tape.operations
+        )
+
+        dev = qml.device("default.qubit")
+        transform_program = dev.preprocess()[0]
+        res1, res2 = qml.execute(
+            [old_tape, new_tape], device=dev, transform_program=transform_program
+        )
+        qml.math.allclose(res1, tape_fn([res2]))
+
+    def test_qnode_decomposition(self):
+        """Test decomposition for the Clifford transform."""
+
+        dev = qml.device("default.qubit")
+
+        def qfunc():
+            qml.CRX(1, wires=[0, 1])
+            qml.ISWAP(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        original_qnode = qml.QNode(qfunc, dev)
+        transfmd_qnode = qml.QNode(clifford_t_decomposition(qfunc), dev)
+
+        res1, res2 = original_qnode(), transfmd_qnode()
+        assert qml.math.allclose(res1, res2)
+
+        assert all(
+            any(
+                (
+                    isinstance(op, gate) or isinstance(getattr(op, "base", None), gate)
+                    for gate in _CLIFFORD_PHASE_GATES
+                )
+            )
+            for op in transfmd_qnode.tape.operations
+        )
+
+    @pytest.mark.parametrize(
+        ("op"), [qml.RX(1.0, wires="a"), qml.U3(1, 2, 3, wires=[1]), qml.PhaseShift(1.0, wires=[2])]
+    )
+    def test_one_qubit_decomposition(self, op):
+        """Test decomposition for the Clifford transform."""
+
+        decomp_ops, global_ops = _one_qubit_decompose(op)
+        decomp_ops = _fuse_global_phases(decomp_ops + [global_ops])
+
+        assert all(
+            any(
+                (
+                    isinstance(op, gate) or isinstance(getattr(op, "base", None), gate)
+                    for gate in _CLIFFORD_PHASE_GATES + [qml.RZ]
+                )
+            )
+            for op in decomp_ops
+        )
+
+        decomp_ops, global_ops = decomp_ops[:-1], decomp_ops[-1]
+        assert isinstance(global_ops, qml.GlobalPhase)
+
+        matrix_op = reduce(
+            lambda x, y: x @ y, [qml.matrix(op) for op in decomp_ops][::-1]
+        ) * qml.matrix(global_ops)
```

Comment:
should it not be this?
```suggestion
        ) @ qml.matrix(global_ops)
```

Reply:
The matrix for `GlobalPhase` is like this `([[phase]])`, so I think the latter would throw an error.